beacon/goclient/signing.go

```diff
@@ -58,7 +58,7 @@ func (gc *GoClient) computeVoluntaryExitDomain(ctx context.Context) (phase0.Doma
 	}
 
 	start = time.Now()
-	genesisResponse, err := gc.multiClient.Genesis(ctx, &api.GenesisOpts{})
+	genesis, err := gc.Genesis(ctx)
 	recordRequestDuration(gc.ctx, "Genesis", gc.multiClient.Address(), http.MethodGet, time.Since(start), err)
 	if err != nil {
 		gc.log.Error(clResponseErrMsg,
```

Comment:
We should move the request duration recording and error logging inside `GoClient.Genesis()` instead of duplicating it in multiple places.

Both `computeVoluntaryExitDomain()` and `getForkInfo()` call `Genesis()`, but only `computeVoluntaryExitDomain()` records request duration and logs errors.
This means that calls from `getForkInfo()` do not log errors or track request duration.

Additionally, we should not record request duration when loading from cache, since that is not an actual network request.

Reply:
well spotted! fixed